src/test/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/resthandler/RestExplainActionIT.kt

```diff
@@ -103,24 +103,50 @@ class RestExplainActionIT : IndexStateManagementRestTestCase() {
-                ManagedIndexMetaData.STEP_START_TIME to "${policyIndexName}_STEP_START_TIME",
-                ManagedIndexMetaData.FAILED_STEP to "${policyIndexName}_FAILED_STEP"
-            )
-        )
         val actual = response.asMap()
-        assertResponseMap(expected, actual)
+
+        assertPredicatesOnMetaData(listOf(
+            policyIndexName to listOf(
+            "index.opendistro.index_state_management.policy_name" to policy.id::equals,
+            ManagedIndexMetaData.INDEX to managedIndexConfig.index::equals,
+            ManagedIndexMetaData.INDEX_UUID to managedIndexConfig.indexUuid::equals,
+            ManagedIndexMetaData.POLICY_NAME to managedIndexConfig.policyName::equals,
+            ManagedIndexMetaData.POLICY_SEQ_NO to policy.seqNo.toInt()::equals,
+            ManagedIndexMetaData.POLICY_PRIMARY_TERM to policy.primaryTerm.toInt()::equals,
+            ManagedIndexMetaData.STATE to policy.defaultState::equals,
+            ManagedIndexMetaData.STATE_START_TIME to fun(startTime: Any?): Boolean = (startTime as Long) < Instant.now().toEpochMilli(),
+            ManagedIndexMetaData.FAILED to false::equals
+        )), actual)
+    }
+
+    fun `test failed policy`() {
+        val policyIndexName = "test_policy_index"
+        val policyName = "does_not_exist"
+        createIndex(policyIndexName, policyName = policyName)
+
+        Thread.sleep(2000)
+
+        val managedIndexConfig = getManagedIndexConfig(policyIndexName)
+        assertNotNull("ManagedIndexConfig is null", managedIndexConfig)
+        // change the start time so the job will trigger in 2 seconds.
+        updateManagedIndexConfigStartTime(managedIndexConfig!!, Instant.now().minusSeconds(58).toEpochMilli())
+
+        Thread.sleep(3000)
+
+        val response = client().makeRequest(RestRequest.Method.GET.toString(), "${RestExplainAction.EXPLAIN_BASE_URI}/$policyIndexName")
+        assertEquals("Unexpected RestStatus.", RestStatus.OK, response.restStatus())
+
+        val expectedInfoString = mapOf("message" to "Could not load policy: $policyName").toString()
+        val actual = response.asMap()
+        assertPredicatesOnMetaData(listOf(
+                policyIndexName to listOf(
+                    "index.opendistro.index_state_management.policy_name" to policyName::equals,
+                    ManagedIndexMetaData.INDEX to managedIndexConfig.index::equals,
+                    ManagedIndexMetaData.INDEX_UUID to managedIndexConfig.indexUuid::equals,
+                    ManagedIndexMetaData.POLICY_NAME to managedIndexConfig.policyName::equals,
+                    ManagedIndexMetaData.FAILED to true::equals,
+                    ManagedIndexMetaData.INFO to fun(info: Any?): Boolean = expectedInfoString == info.toString()
+                )), actual)
     }
 
     @Suppress("UNCHECKED_CAST") // Do assertion of the response map here so we don't have many places to do suppression.
```

Comment:
I don't see function being used anymore. You should remove `assertResponseMap` and `assertMetaDataEntries` if they are no longer being used.

Reply:
The are still being used in the tests at beginning of file.